### PR TITLE
support urlpath in LegacyURLHandler and add testing bootstraps

### PR DESCRIPTION
**Pull request by minrk**
previous binder supported this and there are links in the wild relying on it

### CODE REVIEW

File: binderhub/main.py [hunk 1]
```diff
@@ -49,6 +50,8 @@ def get(self, provider_prefix, spec):
 class LegacyRedirectHandler(BaseHandler):
     """Redirect handler from legacy Binder"""
 
-    def get(self, user, repo):
+    def get(self, user, repo, urlpath=None):
         url = '/v2/gh/{user}/{repo}/master'.format(user=user, repo=repo)
+        if url_path:
```
Comment (betatim) [inline binderhub/main.py]: nit: `if url_path is not None:`

**Comment by betatim** [comment]
LGTM. Test?

I can add a test as well if you point me to an example in jhub's tests that tests something like this.

**Comment by minrk** [comment]
testing handlers is tricky without setting up a binderhub instance, which is tricky without a running kubernetes. I'm working on getting kubernetes deployed on Travis so we can start doing that.

**Comment by betatim** [comment]
Ok. I don't think this feature should be held up by missing test, just thought it would be a good excuse to add some of the scaffolding so we can easily add tests for handlers in the future.

**Comment by minrk** [comment]
I should have a running app for the tests by the time I'm done with #282

**Comment by minrk** [comment]
While working on #282, I realized that I didn't need kubernetes and jupyterhub running to run tests with the app, so I moved that over here and added coverage for this handler.

I have to run, but hopefully these will pass now after chasing down some weird delays

**Comment by minrk** [comment]
This ought to be good to go, then I can rebase and finish #282 to get tests running with a fully armed and operational binderhub setup.

**Comment by choldgraf** [comment]
Looks like a great addition to me! I'll let @betatim give the final merge go-ahead on this.

**Comment by willingc** [comment]
Thanks @minrk.

**Comment by choldgraf** [comment]
Ah I was hoping after @minrk 's "fully armed and operational" comment, somebody would say "you may fire when ready" before merging :-)

**Comment by willingc** [comment]
@choldgraf :smile:

@betatim I am leaving for you to merge into mybinder-deploy in case you see anything else you would like.

**Pull request merged**